docs/concepts/policy/pod-security-policy.md

```diff
@@ -1,97 +1,359 @@
 
-- *MustRunAs* - Requires at least one range to be specified. Uses the
-minimum value of the first range as the default. Validates against the
-first ID in the first range.
-- *RunAsAny* - No default provided. Allows any `fsGroup` ID to be specified.
+Most Kubernetes pods are not created directly by user. Instead, they are
+typically created indirectly as part of a
+[Deployment](/docs/concepts/workloads/controllers/deployment/),
+[ReplicaSet](/docs/concepts/workloads/controllers/replicaset/), or other
+templated controller via the controller manager. Granting the controller access
+to the policy would grant access for *all* pods created by that the controller,
+so the preferable method for authorizing policies is to grant access to the
+pod's service account (see [example](#run-another-pod)).
+
+### Via RBAC
+
+[RBAC](/docs/admin/authorization/rbac/) is a standard Kubernetes authorization
+mode, and can easily be used to authorize use of policies.
+
+First, a `Role` or `ClusterRole` needs to grant access to `use` the desired
+policies. The rules to grant access look like this:
+
+```yaml
+rules:
+- apiGroups: ['extensions']
+  resources: ['podsecuritypolicies']
+  verbs:     ['use']
+  resourceNames:
+  - <list of policies to authorize>
+```
+
+Then the `(Cluster)Role` is bound to the authorized user(s):
+
+```yaml
+subjects:
+# Authorize specific service accounts:
+- kind: ServiceAccount
+  name: <authorized service account name>
+  namespace: <authorized pod namespace>
+# Authorize specific users (not recommended):
+- kind: User
+  apiGroup: rbac.authorization.k8s.io
+  name: <authorized user name>
+```
+
+If a `RoleBinding` (not a `ClusterRoleBinding`) is used, it will only grant
+usage for pods being run in the same namespace as the binding. This can be
+paired with system groups to grant access to all pods run in the namespace:
+```yaml
+# Authorize all service accounts in a namespace:
+- kind: Group
+  apiGroup: rbac.authorization.k8s.io
+  name: system:serviceaccounts
+# Or equivalently, all authenticated users in a namespace:
+- kind: Group
+  apiGroup: rbac.authorization.k8s.io
+  name: system:authenticated
+```
+
+For more examples of RBAC bindings, see [Role Binding
+Examples](docs/admin/authorization/rbac/#role-binding-examples). For a complete
+example of authorizing a PodSecurityPolicy, see
+[below](#example).
+
+## Policy Order
+
+In addition to restricting pod creation and update, pod security policies can
+also be used to provide default values for many of the fields that it
+controls. However, this presents a problem when multiple policies are available
+that provide conflicting defaults. To work around this, the pod security policy
+controller selects policies in the following order:
+
+1. If any policies successfully validate the pod without altering it, they are
+   used.
+2. Otherwise, the first valid policy in alphabetical order is used.
+
+## Example
```

Comment:
I don't see the value in that. Personally I find it much more useful to have a example alongside the documentation, so that I have access to both when I click the first search result.

If you're worried about the length of this page, I could be convinced to separate out the reference section.

Reply:
I'm mostly okay with simple examples e.g. a manifest showing users how things look like.
This "example" is a little bit long. It is actually an exercise for users. They will need to create a separate namespace.
This is just my own feelings. I understand people have different reading habits. :)

Reply:
I'm curious what other's think about this. Is your goal to keep tutorials & reference docs separate, or just to shrink this page? I personally like having a thorough example alongside the reference docs, but we should go for consistency.

Reply:
I like the idea but I'd suggest to postpone it and merge this PR as-is.